### DISCUSSION TITLE
Howto: custom view to display dynamic data from dropdown

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#4644] [Help] [answered]
I am trying to wrap my head around blade, livewire, alpine and what is responsible for what.

Right now I have an invoice model with a n:1 relationship with a customer model.
In the invoice form I want to have a dropdown field to select a customer. Once the user picks a customer the customer's related data (address, telephone and so on) should be displayed in a separate part of the site (not as form fields).

A little similar to the detailed view of an order in the filament demo. On the side there are two dates (via placeholder) in a card. I would like to display the customer's address once the customer is changed in the dropdown.

Any idea how this could be done?

Thank you

**Comment by ManojKiranA** [comment, marked as answer]
you can use `reactive` and `afterStateUpdated` together

Something like this

```
Select::make('customer_id')
    ->reactive()
    ->options([
        //customers list
    ])
    ->afterStateUpdated(function (Closure $set, $state) {
        $set('customer_address', Customer::find($state)?->customer_address);
    })
```

**Reply by OvalMedia** [reply]
Thank you. What does 'customer_address' refer to? I assume this is the target element which is supposed to hold the found data?

**Reply by falconsmilie** [reply]
Maybe the docs help; https://filamentphp.com/docs/2.x/forms/advanced#updates

**Reply by OvalMedia** [reply]
I have already read them. I guess, what I am missing here is the HTML part. What is 'slug' referring to? I would assume the target elements will be addressed via class, id or some other attribute. The docs just mention "the slug field" leaving me in the rain.

**Reply by OvalMedia** [reply]
Ok, I kinda get it. "slug" refers to another form field which has been created in the schema function. But I don't want the address to be displayed in a form field but rather in a custom html element which I injected to the page via a ViewField with a custom template. 

Example result:

```
<address id="customer-address">
Example Company<br />
c/o Jane Doe<br />
Custom Road 123<br />
12345 Atlantis
</address>

```

**Reply by danharrin** [reply]
inside the View, you can use `$evaluate(fn ($get) => $get('select_name'))` to get the value of the select and change the address

**Reply by ManojKiranA** [reply]
> Thank you. What does 'customer_address' refer to? I assume this is the target element which is supposed to hold the found data?

yes `customer_address` wil be another `TextBox` in the same form

**Reply by OvalMedia** [reply]
> inside the View, you can use `$evaluate(fn ($get) => $get('select_name'))` to get the value of the select and change the address

Sorry, I don't follow. Where do I put that code?

**Reply by danharrin** [reply]
`View::make('filament.forms.customer-address')`

resources/views/filament/forms/customer-address.blade.php:

```blade
{{ $evaluate(fn ($get) => $get('select_name')) }}
```

**Reply by OvalMedia** [reply]
Sorry, but this gets even more confusing. I assume this is all still too high for me at this stage.

**Reply by OvalMedia** [reply]
Ok, when I do this, all that happens is that the value (client id in this case) of the named select will be displayed on the spot.
How do I get and display the related data of the selected client?

**Reply by danharrin** [reply]
`Client::find(client ID here)`, use Eloquent as you would normally

**Comment by OvalMedia** [comment]
For the sake of completeness:

`<address>
{!! $evaluate(fn ($get) => App\Models\Client::find($get('client_id')))->addressAsHtml() !!}
</address>`

`addressAsHtml` is a function that concatenates all address field of that model and returns is as html.
